### ISSUE TITLE
File-Line is omitted when suggesting possible candidates depending on noMatchCount

### ISSUE THREAD

**Issue by ferdymercury** [doxygen/doxygen#9276]
**Describe the bug**
When suggesting "possible candidates", the file and line is omitted if there are no "uniquely" matching ones.

**Expected behavior**
Possible candidates printed include file-line.

**Screenshots**

```
/opt/root_src/core/metacling/src/TClingClassInfo.cxx:507: warning: no matching class member found for 
  TClingMethodInfo TClingClassInfo::GetMethodWithArgs(const char *fname, const char *arglist, bool objectIsConst, Longptr_t *poffset, EFunctionMatchMode, EInheritanceMode) const
Possible candidates:
  'TClingMethodInfo TClingClassInfo::GetMethodWithArgs(const char *fname, const char *arglist, Longptr_t *poffset, ROOT::EFunctionMatchMode mode=ROOT::kConversionMatch, EInheritanceMode imode=kWithInheritance) const'
  'TClingMethodInfo TClingClassInfo::GetMethodWithArgs(const char *fname, const char *arglist, bool objectIsConst, Longptr_t *poffset, ROOT::EFunctionMatchMode mode=ROOT::kConversionMatch, EInheritanceMode imode=kWithInheritance) const'
```

vs

```
/opt/root_src/core/metacling/src/TClingClassInfo.cxx:752: warning: no uniquely matching class member found for 
  void TClingClassInfo::Init(const Decl *decl)
Possible candidates:
  'void TClingClassInfo::Init(const char *name)' /opt/root_src/core/metacling/src/TClingClassInfo.h:165
  'void TClingClassInfo::Init(const clang::Decl *)' /opt/root_src/core/metacling/src/TClingClassInfo.h:166
  'void TClingClassInfo::Init(int tagnum)' /opt/root_src/core/metacling/src/TClingClassInfo.h:167
  'void TClingClassInfo::Init(const clang::Type &)' /opt/root_src/core/metacling/src/TClingClassInfo.h:168
```

**To Reproduce**
[candidates.zip](https://github.com/doxygen/doxygen/files/8484112/candidates.zip)

**Version**
1.9.4 (e449f7367f5b7f6efadba28747dba591e1120bc5*)

**Additional context**
https://github.com/doxygen/doxygen/blob/master/src/doxygen.cpp#L6039
https://github.com/doxygen/doxygen/issues/9143#issuecomment-1098031112

Before https://github.com/doxygen/doxygen/pull/7312 the else part was completely missing.
The if (noMatchCount>1) part was added between the versions 1.4.7 and 1.5.0 (more precisely in the 1.4.7_20061004 version) but I could not find the reason / bug report for the change and only in case if (noMatchCount>1).

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9277

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
